forui/lib/src/widgets/accordion/accordion.dart

```diff
@@ -0,0 +1,220 @@
+
+import 'package:forui/forui.dart';
+
+/// A vertically stacked set of interactive headings that each reveal a section of content.
+///
+/// Typically used to group multiple [FAccordionItem]s.
+///
+/// See:
+/// * https://forui.dev/docs/FAccordion for working examples.
+/// * [FAccordionController] for customizing the accordion's selection behavior.
+/// * [FAccordionStyle] for customizing a select group's appearance.
+class FAccordion extends StatefulWidget {
+  /// The controller.
+  ///
+  /// See:
+  /// * [FAccordionController] for default multiple selections.
+  /// * [FAccordionController.radio] for a radio-like selection.
+  final FAccordionController? controller;
+
+  /// The items.
+  final List<FAccordionItem> children;
+
+  /// The style. Defaults to [FThemeData.accordionStyle].
+  final FAccordionStyle? style;
+
+  /// Creates a [FAccordion].
+  const FAccordion({
+    required this.children,
+    this.controller,
+    this.style,
+    super.key,
+  });
+
+  @override
+  State<FAccordion> createState() => _FAccordionState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('controller', controller))
+      ..add(DiagnosticsProperty('style', style))
+      ..add(IterableProperty('items', children));
+  }
+}
+
+class _FAccordionState extends State<FAccordion> {
+  late FAccordionController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = widget.controller ?? FAccordionController();
+
+    if (!_controller.validate(widget.children.where((child) => child.initiallyExpanded).length)) {
+      throw StateError('number of expanded items must be within the allowed range.');
+    }
+  }
+
+  @override
+  void didUpdateWidget(covariant FAccordion oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    if (widget.controller != oldWidget.controller) {
+      _controller = widget.controller ?? FAccordionController();
+
+      if (!_controller.validate(widget.children.where((child) => child.initiallyExpanded).length)) {
+        throw StateError('number of expanded items must be within the min and max.');
+      }
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) => Column(
+        children: [
+          for (final (index, child) in widget.children.indexed)
+            FAccordionItemData(
+              index: index,
+              controller: _controller,
+              child: child,
+            ),
+        ],
+      );
+}
+
+/// The [FAccordion] style.
+final class FAccordionStyle with Diagnosticable {
+  /// The title's text style.
+  final TextStyle titleTextStyle;
+
+  /// The child's default text style.
+  final TextStyle childTextStyle;
+
+  /// The padding around the title.
+  final EdgeInsets titlePadding;
+
+  /// The padding around the content.
+  final EdgeInsets childPadding;
+
+  /// The icon.
+  final SvgPicture icon;
```

Comment:
@kawaijoe should we use `SvgPicture` as the type?

Reply:
I think it could be better to accept any widget.